### PR TITLE
use latin-1 encoding when reading/writing .idl files, prepend BOM to generated C/C++ files when necessary

### PR DESCRIPTION
**Pull request by dirk-thomas**
Addresses the problem on Windows related to ros2/test_interface_files#8.

[Build Status](https://ci.ros2.org/job/ci_windows/7438/) (only passed because these tests are disabled: ros2/system_tests#377).

### CODE REVIEW

**Comment by dirk-thomas** [comment]
After all the issues in the pipeline have been addressed the second commit in this PR fixes the actual problem from ros2/rosidl#388 and prepends a BOM to generated C/C++ if necessary: [Build Status](https://ci.ros2.org/job/ci_windows/7470/)

File: rosidl_generator_c/rosidl_generator_c/__init__.py [hunk 1]
```diff
@@ -34,7 +34,20 @@ def generate_c(generator_arguments_file):
...
+    generate_files(
+        generator_arguments_file, mapping,
+        post_process_callback=prefix_with_bom_if_necessary)
+
+
+def prefix_with_bom_if_necessary(content):
+    try:
+        content.encode('ASCII')
+    except UnicodeError:
+        prefix = '\ufeff' + \
```
Comment (sloretz) [inline rosidl_generator_c/rosidl_generator_c/__init__.py]: [2.13 Page 58 (pdf page 52)](https://www.unicode.org/versions/Unicode5.0.0/ch02.pdf) says the utf-8 BOM is `0xEF, 0xBB, 0xBF`. This looks like the UTF-16 BOM.

File: rosidl_generator_cpp/rosidl_generator_cpp/__init__.py [hunk 2]
```diff
@@ -34,7 +34,20 @@ def generate_cpp(generator_arguments_file):
...
+    generate_files(
+        generator_arguments_file, mapping,
+        post_process_callback=prefix_with_bom_if_necessary)
+
+
+def prefix_with_bom_if_necessary(content):
+    try:
+        content.encode('ASCII')
+    except UnicodeError:
+        prefix = '\ufeff' + \
```
Comment (sloretz) [inline rosidl_generator_cpp/rosidl_generator_cpp/__init__.py]: same comment here

File: rosidl_generator_c/rosidl_generator_c/__init__.py (hunk 1 above)
Comment (dirk-thomas) [reply rosidl_generator_c/rosidl_generator_c/__init__.py]: I just replicated what we already use in the only existing file with a BOM: https://github.com/ros2/rcl_interfaces/blob/75e94e1d10399fda56ccb7c635aef730e3d7def0/test_msgs/include/test_msgs/message_fixtures.hpp#L1

`hexdump -n 3 -C <filename>` returns `ef bb bf`

Comment (sloretz) [reply rosidl_generator_c/rosidl_generator_c/__init__.py]: Ah oops, I get it. `\ufeff` is the code point, and `ef bb bf` are the encoded bytes. 

```python
>>> u"\ufeff".encode('utf-8')
'\xef\xbb\xbf'
```

**Pull request merged**